tests/tests_caching_logic.py

```diff
@@ -47,24 +51,35 @@ async def test_num_infer_calls(self):
       # no service calls before executing query
```

Comment:
We should have a separate test that tests that other ML models don't get removed when one is removed